contracts/EtomicSwapV2.sol

```diff
@@ -0,0 +1,481 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.20;
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
+contract EtomicSwapV2 {
+    enum MakerPaymentState {
+        Uninitialized,
+        PaymentSent,
+        TakerSpent,
+        MakerRefunded
+    }
+
+    struct MakerPayment {
+        bytes20 paymentHash;
+        uint32 paymentLockTime;
+        MakerPaymentState state;
+    }
+
+    event MakerPaymentSent(bytes32 id);
+    event MakerPaymentSpent(bytes32 id, bytes32 secret);
+    event MakerPaymentRefundedTimelock(bytes32 id);
+    event MakerPaymentRefundedSecret(bytes32 id, bytes32 secret);
+
+    mapping(bytes32 => MakerPayment) public maker_payments;
+
+    enum TakerPaymentState {
+        Uninitialized,
+        PaymentSent,
+        TakerApproved,
+        MakerSpent,
+        TakerRefunded
+    }
+
+    struct TakerPayment {
+        bytes20 paymentHash;
+        uint32 immediateRefundTime;
+        uint32 paymentLockTime;
+        TakerPaymentState state;
+    }
+
+    event TakerPaymentSent(bytes32 id);
+    event TakerPaymentApproved(bytes32 id);
+    event TakerPaymentSpent(bytes32 id, bytes32 secret);
+    event TakerPaymentRefundedSecret(bytes32 id, bytes32 secret);
+    event TakerPaymentRefundedTimelock(bytes32 id);
+
+    mapping(bytes32 => TakerPayment) public taker_payments;
+
+    address public dexFeeAddress;
```

Comment:
declare `dexFeeAddress` as immutable

```solidity
address public immutable dexFeeAddress;
```

This adjustment will make contract more gas-efficient when `dexFeeAddress` is accessed and also signals to others that this value is constant post-deployment, improving the contract's readability and security posture.

Reply:
Good note, thanks :slightly_smiling_face: